src/web/resources/report.js

```diff
@@ -35,6 +35,100 @@ function Element(tagname, props, children) {
     return $elt;
 }
 
+const renames = {
+    "imp-start": "Improved start",
+    "apx-start": "Approximate start",
+    "uni-start": "Regressed from start",
+    "ex-start": "Exact start",
+    "eq-start": "Equal start",
+    "lt-start": "Less than start",
+    "gt-start": "Greater than start",
+    "gt-target": "Greater than target",
+    "eq-target": "Equal than target",
+    "lt-target": "Less than target",
+    "error": "Error",
+    "timeout": "Timeout",
+    "crash": "Crash",
+}
+
+const Results = new Component("#results", {
+    setup: function () {
+        // clickable rows
+        let $rows = this.elt.querySelectorAll("tbody tr");
+        for (let $row of $rows) {
+            $row.addEventListener("click", () => $row.querySelector("a").click());
+        }
+
+        this.setupFilters()
+    },
+    setupFilters: function () {
+        const regressedTags = ["uni-start", "lt-target", "lt-start",
+            "apx-start", "timeout", "crash", "error"]
+        const improvedTags = ["imp-start", "ex-start", "eq-start", "eq-target",
+            "gt-target"]
+
+        const improvedChildren = improvedTags.map((child) => {
+            const count = this.getRowsForClass(child).length
+            return this.createChild(child, count)
+        })
+        const regressedChildren = regressedTags.map((child) => {
+            const count = this.getRowsForClass(child).length
+            return this.createChild(child, count)
+        })
+
+        // add listeners
+        const improvedLeader = this.attachLeaderToChildren("improved","Improved", improvedChildren)
+        const regressedLeader = this.attachLeaderToChildren("regressed","Regressed", regressedChildren)
+
+        this.elt.parentNode.insertBefore(Element("div", { id: "filters" }, [
+            Element("div", { classList: "section-title" }, "Filters"),
+            Element("div", { id: "filter-group" }, [
+                improvedLeader, regressedLeader]),
+            Element("details", [
+                Element("summary", "Advanced"), [
+                    improvedChildren, regressedChildren]])]), this.elt)
+    },
+    attachLeaderToChildren: function (leaderTag,leaderName, childNodes) {
+        const parentLabel = this.buildCheckboxLabel(leaderTag, leaderName, true)
+        parentLabel.addEventListener("click", () => {
+            const parentState = parentLabel.querySelector("input").checked
+            childNodes.forEach((child) => {
+                const children = this.getRowsForClass(child.classList[0])
```

Comment:
This is bad, as it assumes that there is at least one class and that the first one will be the selector for the child rows. Would it be better to move the map that creates the childFilter labels into this function and do all the work inside attachLeaderToChildren and rename the function?